### PR TITLE
Fix email validation

### PR DESCRIPTION
**Pull request by ZakarFin**
Seems that the email validation was broken in #924. This fixes the issue and uses regexp from https://owasp.org/www-community/OWASP_Validation_Regex_Repository for validating the address.

Note to self: add/require more unit tests...

### CODE REVIEW

File: control-users/src/main/java/fi/nls/oskari/control/users/RegistrationUtil.java [hunk 1]
```diff
@@ -13,6 +13,9 @@
  */
 public class RegistrationUtil {
 
+    // From: https://owasp.org/www-community/OWASP_Validation_Regex_Repository
+    private static final String EMAIL_REGEXP = "^[a-zA-Z0-9_+&*-]+(?:\\.[a-zA-Z0-9_+&*-]+)*@(?:[a-zA-Z0-9-]+\\.)+[a-zA-Z]{2,}$";
+
```
Comment (jampukka) [inline control-users/src/main/java/fi/nls/oskari/control/users/RegistrationUtil.java:18]:
```suggestion
    private static final String EMAIL_PATTERN = Pattern.compile(EMAIL_REGEXP);
```

File: control-users/src/main/java/fi/nls/oskari/control/users/RegistrationUtil.java [hunk 2]
```diff
@@ -23,8 +26,7 @@ public static final String getServerAddress(ActionParameters params) {
     }
 
     public static boolean isValidEmail(String email) {
-        String regex = "^[\\w-_\\.+]*[\\w-_\\.]\\@([\\w]+\\.\\-)+[\\w]+[\\w]$";
-        return email != null && !email.isEmpty() && email.matches(regex);
+        return email != null && !email.isEmpty() && email.matches(EMAIL_REGEXP);
```
Comment (jampukka) [inline control-users/src/main/java/fi/nls/oskari/control/users/RegistrationUtil.java:29]:
```suggestion
        return email != null && !email.isEmpty() && EMAIL_PATTERN.matcher(email).matches();;
```

**Comment by jampukka** [comment]
Uhm sorry, I wanted test the suggest changes but ended up creating commits to the PR, whoopsie!

**Comment by ZakarFin** [comment]
No worries :) Thanks for the review :D

**Pull request merged**